### ISSUE TITLE
Redshift: RF03 rule violations with where exists

### ISSUE THREAD

**Issue by maoxingda** [sqlfluff/sqlfluff#5599]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
insert into dim.clients
(
    data_source
)
select s.data_source
from
    dim.clients_stg as s
where
    not exists (
        select 1 from dim.clients as t
        where s.id = t.id
    )
;
```

**Rule L026 finds violations in where exists**

### Expected Behaviour

No RF03 rule violation.

### Observed Behaviour

```
L:  11 | P:  15 | RF03 | Unqualified reference 's.id' found in single table select. [references.consistent]
```

### How to reproduce

```
sqlfluff lint test.sql
```

### Dialect

redshift

### Version

sqlfluff, version 3.0.0a4

### Configuration

```cfg
[sqlfluff]
dialect = redshift
warn_unused_ignores = True
large_file_skip_byte_limit = 0
max_line_length = 256
output_line_length = 256

rules =
    references.consistent

# warnings =
#     RF04

[sqlfluff:rules]
single_table_references = qualified

[sqlfluff:rules:references.consistent]
force_enable = True
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by raginjason** [comment]
I am having the same issue with v3.0.3
